lib/rules/use-tokens/__tests__/index.test.js

```diff
@@ -0,0 +1,187 @@
+/*
+ Copyright 2020 Skyscanner
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+const { lint } = require('stylelint');
+
+const config = {
+  plugins: ['./index.js'],
+  rules: {
+    'backpack/use-tokens': [true],
+  },
+};
+
+describe('Spacing token tests', () => {
+  it('passes when setting spacing', async () => {
+    const {
+      results: [{ warnings }],
+    } = await lint({
+      code: 'a { width: $bpk-heading-content-margin-top; font-size: 14rem }',
+      config,
+    });
+
+    expect(warnings).toHaveLength(0);
+  });
+
+  it('should not throw as it uses a function', async () => {
```

Comment:
Is this a conscious decision or a limitation? I think `-1px` shouldn't be a valid usage as it breaks the spacing grid.

Reply:
True but when it comes to functions such as `calc` as used in this test we can't determine the result or the values being used so we have to exclude it